src/commentmodel.cpp

```diff
@@ -541,8 +541,11 @@ void CommentModel::moreComments(int index, const QVariant &children)
     QHash<QString, QString> parameters;
     parameters["sort"] = getSortString(m_sort);
     parameters["link_id"] = linkMap.value("fullname").toString();
-    QStringList limitedChildrenList(children.toStringList().mid(0,50));
-    m_moremoreComments = children.toStringList().mid(50);
+    QStringList const childrenList = children.toStringList();
+    QStringList const limitedChildrenList = childrenList.mid(0,50);
+    if (childrenList.size() > 50) {
+        m_moremoreComments = childrenList.mid(50);
+    }
```

Comment:
this also needs an else clause, setting the m_moremoreComments to an empty list if there's <= 50 items, otherwise you can leave state from a previous moreComments call.

Reply:
This is the only setter for this variable. It either gets in or stays default-initialized (empty).

Reply:
Thanks, done!